taipy/gui_core/_context.py

```diff
@@ -1044,9 +1044,9 @@ def update_data(self, state: State, id: str, payload: t.Dict[str, str]):
                     else float(val)
                     if data.get("type") == "float"
                     else data.get("value"),
+                    editor_id=self.gui._get_client_id(),
```

Comment:
@FredLL-Avaiga, I believe this answers the comment you made a few days ago on another PR.

Reply:
it does indeed

Reply:
we could use a public API  `get_state_id(state)` but `self.gui._get_client_id()` is already used when state is not available